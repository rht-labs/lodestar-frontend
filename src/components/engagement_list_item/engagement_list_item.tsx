import React from 'react';
import {
  Button,
  Card,
  CardBody,
  Title,
  Grid,
  GridItem,
  Label,
} from '@patternfly/react-core';
import { useHistory } from 'react-router';
import { EngagementAtAGlance } from './engagement_at_a_glance';
import { getEngagementStatus, Engagement } from '../../schemas/engagement';

export interface DataCardProps {
  engagement: Engagement;
}

export function EngagementListItem(props: DataCardProps) {
  const { engagement } = props;
  const url = `/app/engagements/${engagement.customer_name}/${engagement.project_name}`;
  const history = useHistory();
  const status = getEngagementStatus(engagement);

  return (
    <>
      <Card isCompact style={{ margin: '1rem', padding: '2rem 0.5rem' }}>
        <CardBody style={{ padding: '0 2rem' }}>
          <Grid hasGutter style={{ alignItems: 'center' }}>
            <GridItem md={12} lg={3}>
              <Button
                variant="link"
                isInline
                style={{
                  whiteSpace: 'normal',
                  wordWrap: 'break-word',
                  textAlign: 'left',
                }}
                onClick={() => {
                  history.push(url);
                }}
              >
                <Title headingLevel="h3" style={{ fontWeight: 'bolder' }}>
                  {engagement.project_name}
                </Title>
              </Button>
              <Title style={{ fontWeight: 'normal' }} headingLevel="h4">
                {engagement.customer_name}
              </Title>
              {engagement.engagement_categories?.length > 0 ? (
                engagement?.engagement_categories.map(currentChip => (
                  <div key={currentChip.name}>
                    <Label
                      key={currentChip.name}
                      style={{ marginTop: '1rem', marginRight: '0.5rem' }}
                      color="blue"
                    >
                      {currentChip.name}
                    </Label>
                  </div>
                ))
              ) : (
                <></>
              )}
            </GridItem>
            <GridItem md={12} lg={7}>
              <EngagementAtAGlance engagement={engagement} status={status} />
            </GridItem>
            <GridItem sm={12} md={4} lg={2}>
              <Button
                onClick={() => {
                  history.push(url);
                }}
                variant="secondary"
                data-cy={'view_engagement_button'}
              >
                View Engagement
              </Button>
            </GridItem>
          </Grid>
        </CardBody>
      </Card>
    </>
  );
}
