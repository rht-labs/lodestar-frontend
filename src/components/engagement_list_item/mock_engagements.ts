import { Engagement } from '../../schemas/engagement';
import { HealthStatus } from '../../schemas/cluster_status';

export function upcomingEngagement(): Engagement {
  return {
    archive_date: new Date(),
    customer_contact_email: 'bob@doe.com',
    customer_contact_name: `Bob Doe`,
    customer_name: 'NASA',
    description: "It's rocket science",
    end_date: new Date(2050, 1, 3),
    engagement_users: [],
    engagement_lead_email: 'alice@doe.com',
    engagement_lead_name: `Alice Doe`,
    location: `Nashville, TN`,
    ocp_cloud_provider_name: 'AWS',
    ocp_cloud_provider_region: 'N. Virginia',
    ocp_cluster_size: 'Large',
    commits: [],
    creation_details: {
      created_by_email: 'john@doe.com',
      created_by_user: 'John Doe',
      created_on: new Date(2020, 1, 1),
    },
    status: {
      messages: [],
      overall_status: HealthStatus.green,
      subsystems: [],
    },
    ocp_persistent_storage_size: `100G`,
    ocp_sub_domain: 'ordoabchao',
    ocp_version: '4.4',
    project_id: 1,
    project_name: 'Boots on the Moon',
    start_date: new Date(2050, 1, 1),
    technical_lead_email: 'eve@doe.com',
    technical_lead_name: `Eve Doe`,
    launch: null,
  };
}

export function currentEngagement(): Engagement {
  return {
    archive_date: new Date(),
    customer_contact_email: 'bob@doe.com',
    customer_contact_name: `Bob Doe`,
    customer_name: 'NASA',
    description: "It's rocket science",
    end_date: new Date(2050, 1, 1),
    engagement_users: ['a', 'b', 'c'],
    engagement_lead_email: 'alice@doe.com',
    engagement_lead_name: `Alice Doe`,
    location: `Nashville, TN`,
    ocp_cloud_provider_name: 'AWS',
    ocp_cloud_provider_region: 'N. Virginia',
    ocp_cluster_size: 'Large',
    commits: [],
    creation_details: {
      created_by_email: 'john@doe.com',
      created_by_user: 'John Doe',
      created_on: new Date(2020, 1, 1),
    },
    status: {
      messages: [],
      overall_status: HealthStatus.green,
      subsystems: [],
    },
    ocp_persistent_storage_size: `100G`,
    ocp_sub_domain: 'ordoabchao',
    ocp_version: '4.4',
    project_id: 1,
    project_name: 'Boots on the Moon',
    start_date: new Date(2000, 1, 1),
    technical_lead_email: 'eve@doe.com',
    technical_lead_name: `Eve Doe`,
    launch: {
      launched_by: 'John Doe',
      launched_date_time: new Date(2020, 3, 21),
    },
  };
}

export function pastEngagement(): Engagement {
  return {
    archive_date: new Date(),
    customer_contact_email: 'bob@doe.com',
    customer_contact_name: `Bob Doe`,
    customer_name: 'NASA',
    description: "It's rocket science",
    end_date: new Date(2020, 1, 3),
    engagement_users: ['1', '2', '3', '1', '2', '3'],
    engagement_lead_email: 'alice@doe.com',
    engagement_lead_name: `Alice Doe`,
    location: `Nashville, TN`,
    ocp_cloud_provider_name: 'AWS',
    ocp_cloud_provider_region: 'N. Virginia',
    ocp_cluster_size: 'Large',
    commits: [],
    creation_details: {
      created_by_email: 'john@doe.com',
      created_by_user: 'John Doe',
      created_on: new Date(2020, 1, 1),
    },
    status: {
      messages: [],
      overall_status: HealthStatus.green,
      subsystems: [],
    },
    ocp_persistent_storage_size: `100G`,
    ocp_sub_domain: 'ordoabchao',
    ocp_version: '4.4',
    project_id: 1,
    project_name: 'Boots on the Moon',
    start_date: new Date(2020, 1, 1),
    technical_lead_email: 'eve@doe.com',
    technical_lead_name: `Eve Doe`,
    launch: {
      launched_by: 'John Doe',
      launched_date_time: new Date(2020, 1, 2),
    },
  };
}
