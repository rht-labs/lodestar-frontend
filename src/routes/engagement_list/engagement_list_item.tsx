import {
  Engagement,
  getEngagementStatus,
} from '../../schemas/engagement_schema';
import React from 'react';
import { EngagementListItemCard } from './engagement_list_item_card';
import { EngagementAtAGlance } from './engagement_at_a_glance';

export function EngagementListItem({ engagement }: { engagement: Engagement }) {
  const status = getEngagementStatus(engagement);
  return (
    <>
      <EngagementListItemCard
        title={engagement.project_name}
        customer={engagement.customer_name}
        project={engagement.project_name}
      >
        <EngagementAtAGlance engagement={engagement} status={status} />
      </EngagementListItemCard>
    </>
  );
}
