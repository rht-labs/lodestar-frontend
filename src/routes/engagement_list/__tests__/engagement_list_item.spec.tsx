import React from 'react';
import { cleanup, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import {
  upcomingEngagement,
  currentEngagement,
  pastEngagement,
} from '../mock_engagements';
import {
  EngagementStatus,
  getEngagementStatus,
  Engagement,
} from '../../../schemas/engagement';
import { EngagementListItem } from '../engagement_list_item';
import { MemoryRouter } from 'react-router';

afterEach(cleanup);

describe('Engagement status', () => {
  test('renders the upcoming status', () => {
    expect(getEngagementStatus(upcomingEngagement())).toEqual(
      EngagementStatus.upcoming
    );
  });
  test('renders the active status', () => {
    expect(getEngagementStatus(currentEngagement())).toEqual(
      EngagementStatus.active
    );
  });
  test('renders the past status', () => {
    expect(getEngagementStatus(pastEngagement())).toEqual(
      EngagementStatus.past
    );
  });

  test('matches snapshot', () => {
    expect(
      render(
        <MemoryRouter>
          <EngagementListItem engagement={Engagement.fromFake(true)} />
        </MemoryRouter>
      )
    ).toMatchSnapshot();
  });
});
