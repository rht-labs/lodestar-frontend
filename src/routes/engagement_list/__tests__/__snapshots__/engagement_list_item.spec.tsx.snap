// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Engagement status matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <article
        class="pf-c-card pf-m-compact"
        data-ouia-component-id="0"
        data-ouia-component-type="PF4/Card"
        data-ouia-safe="true"
        style="margin: 1rem; padding: 2rem 0.5rem;"
      >
        <div
          class="pf-c-card__body"
          style="padding: 0px 2rem;"
        >
          <div
            class="pf-l-grid pf-m-gutter"
            style="align-items: center;"
          >
            <div
              class="pf-l-grid__item pf-m-12-col-on-md pf-m-3-col-on-lg"
            >
              <button
                aria-disabled="false"
                class="pf-c-button pf-m-link pf-m-inline"
                data-ouia-component-id="1"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                style="white-space: normal; word-wrap: break-word; text-align: left;"
                type="button"
              >
                <h3
                  class="pf-c-title pf-m-lg"
                  style="font-weight: bolder;"
                >
                  Boots on the Moon
                </h3>
              </button>
              <h4
                class="pf-c-title pf-m-md"
                style="font-weight: normal;"
              >
                NASA
              </h4>
            </div>
            <div
              class="pf-l-grid__item pf-m-12-col-on-md pf-m-7-col-on-lg"
            >
              <div
                class="pf-l-grid pf-m-gutter"
              >
                <div
                  class="pf-l-grid__item pf-m-6-col-on-md pf-m-12-col-on-lg"
                >
                  <div
                    class="pf-l-grid pf-m-gutter"
                  >
                    <div
                      class="pf-l-grid__item pf-m-12-col"
                    >
                      Created by: 
                      dwasinge
                    </div>
                    <div
                      class="pf-l-grid__item"
                    >
                      Target start date:
                       
                      Jun 01, 2020
                    </div>
                  </div>
                </div>
                <div
                  class="pf-l-grid__item pf-m-12-col-on-md pf-m-6-col-on-lg"
                >
                  <div
                    class="pf-l-flex"
                  >
                    <div
                      class="pf-l-flex"
                    >
                      <div
                        class="pf-m-spacer-sm"
                      >
                        <svg
                          aria-describedby="pf-tooltip-1"
                          aria-hidden="true"
                          fill="currentColor"
                          height="1em"
                          role="img"
                          style="vertical-align: -0.125em;"
                          viewBox="0 0 448 512"
                          width="1em"
                        >
                          <path
                            d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
                            transform=""
                          />
                        </svg>
                      </div>
                      <div
                        class=""
                      >
                        1
                      </div>
                    </div>
                    <div
                      class="pf-l-flex"
                    >
                      <div
                        class="pf-m-spacer-sm"
                      >
                        <svg
                          aria-describedby="pf-tooltip-2"
                          aria-hidden="true"
                          fill="currentColor"
                          height="1em"
                          role="img"
                          style="vertical-align: -0.125em;"
                          viewBox="0 0 512 512"
                          width="1em"
                        >
                          <path
                            d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200zm61.8-104.4l-84.9-61.7c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h32c6.6 0 12 5.4 12 12v141.7l66.8 48.6c5.4 3.9 6.5 11.4 2.6 16.8L334.6 349c-3.9 5.3-11.4 6.5-16.8 2.6z"
                            transform=""
                          />
                        </svg>
                      </div>
                      <div
                        class=""
                      >
                        4
                      </div>
                    </div>
                    <div
                      class="pf-l-flex"
                    >
                      <div
                        class=""
                      >
                        <b
                          data-cy="upcoming"
                          style="color: rgb(236, 122, 8);"
                        >
                          UPCOMING
                        </b>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="pf-l-grid__item pf-m-12-col-on-md pf-m-6-col-on-lg"
                >
                  <p>
                    <span>
                      Updated by 
                    </span>
                    <b>
                      random-user@example.com
                       
                    </b>
                    <b>
                      215 days ago
                    </b>
                  </p>
                </div>
              </div>
            </div>
            <div
              class="pf-l-grid__item pf-m-12-col-on-sm pf-m-4-col-on-md pf-m-2-col-on-lg"
            >
              <button
                aria-disabled="false"
                class="pf-c-button pf-m-secondary"
                data-cy="view_engagement_button"
                data-ouia-component-id="2"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                type="button"
              >
                View Engagement
              </button>
            </div>
          </div>
        </div>
      </article>
    </div>
  </body>,
  "container": <div>
    <article
      class="pf-c-card pf-m-compact"
      data-ouia-component-id="0"
      data-ouia-component-type="PF4/Card"
      data-ouia-safe="true"
      style="margin: 1rem; padding: 2rem 0.5rem;"
    >
      <div
        class="pf-c-card__body"
        style="padding: 0px 2rem;"
      >
        <div
          class="pf-l-grid pf-m-gutter"
          style="align-items: center;"
        >
          <div
            class="pf-l-grid__item pf-m-12-col-on-md pf-m-3-col-on-lg"
          >
            <button
              aria-disabled="false"
              class="pf-c-button pf-m-link pf-m-inline"
              data-ouia-component-id="1"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              style="white-space: normal; word-wrap: break-word; text-align: left;"
              type="button"
            >
              <h3
                class="pf-c-title pf-m-lg"
                style="font-weight: bolder;"
              >
                Boots on the Moon
              </h3>
            </button>
            <h4
              class="pf-c-title pf-m-md"
              style="font-weight: normal;"
            >
              NASA
            </h4>
          </div>
          <div
            class="pf-l-grid__item pf-m-12-col-on-md pf-m-7-col-on-lg"
          >
            <div
              class="pf-l-grid pf-m-gutter"
            >
              <div
                class="pf-l-grid__item pf-m-6-col-on-md pf-m-12-col-on-lg"
              >
                <div
                  class="pf-l-grid pf-m-gutter"
                >
                  <div
                    class="pf-l-grid__item pf-m-12-col"
                  >
                    Created by: 
                    dwasinge
                  </div>
                  <div
                    class="pf-l-grid__item"
                  >
                    Target start date:
                     
                    Jun 01, 2020
                  </div>
                </div>
              </div>
              <div
                class="pf-l-grid__item pf-m-12-col-on-md pf-m-6-col-on-lg"
              >
                <div
                  class="pf-l-flex"
                >
                  <div
                    class="pf-l-flex"
                  >
                    <div
                      class="pf-m-spacer-sm"
                    >
                      <svg
                        aria-describedby="pf-tooltip-1"
                        aria-hidden="true"
                        fill="currentColor"
                        height="1em"
                        role="img"
                        style="vertical-align: -0.125em;"
                        viewBox="0 0 448 512"
                        width="1em"
                      >
                        <path
                          d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
                          transform=""
                        />
                      </svg>
                    </div>
                    <div
                      class=""
                    >
                      1
                    </div>
                  </div>
                  <div
                    class="pf-l-flex"
                  >
                    <div
                      class="pf-m-spacer-sm"
                    >
                      <svg
                        aria-describedby="pf-tooltip-2"
                        aria-hidden="true"
                        fill="currentColor"
                        height="1em"
                        role="img"
                        style="vertical-align: -0.125em;"
                        viewBox="0 0 512 512"
                        width="1em"
                      >
                        <path
                          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200zm61.8-104.4l-84.9-61.7c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h32c6.6 0 12 5.4 12 12v141.7l66.8 48.6c5.4 3.9 6.5 11.4 2.6 16.8L334.6 349c-3.9 5.3-11.4 6.5-16.8 2.6z"
                          transform=""
                        />
                      </svg>
                    </div>
                    <div
                      class=""
                    >
                      4
                    </div>
                  </div>
                  <div
                    class="pf-l-flex"
                  >
                    <div
                      class=""
                    >
                      <b
                        data-cy="upcoming"
                        style="color: rgb(236, 122, 8);"
                      >
                        UPCOMING
                      </b>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="pf-l-grid__item pf-m-12-col-on-md pf-m-6-col-on-lg"
              >
                <p>
                  <span>
                    Updated by 
                  </span>
                  <b>
                    random-user@example.com
                     
                  </b>
                  <b>
                    215 days ago
                  </b>
                </p>
              </div>
            </div>
          </div>
          <div
            class="pf-l-grid__item pf-m-12-col-on-sm pf-m-4-col-on-md pf-m-2-col-on-lg"
          >
            <button
              aria-disabled="false"
              class="pf-c-button pf-m-secondary"
              data-cy="view_engagement_button"
              data-ouia-component-id="2"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              type="button"
            >
              View Engagement
            </button>
          </div>
        </div>
      </div>
    </article>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
